StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Saved Payment Method Screen/SavedPaymentOptionsViewController.swift

```diff
@@ -422,38 +433,38 @@ extension SavedPaymentOptionsViewController: PaymentOptionCellDelegate {
     }
 
     private func removePaymentMethod(paymentOptionCell: SavedPaymentMethodCollectionView.PaymentOptionCell) {
-            guard let indexPath = collectionView.indexPath(for: paymentOptionCell),
-                  case .saved(let paymentMethod) = viewModels[indexPath.row]
-            else {
-                assertionFailure()
-                return
-            }
-            let viewModel = viewModels[indexPath.row]
-            self.viewModels.remove(at: indexPath.row)
-            // the deletion needs to be in a performBatchUpdates so we make sure it is completed
-            // before potentially leaving edit mode (which triggers a reload that may collide with
-            // this deletion)
-            self.collectionView.performBatchUpdates {
-                self.collectionView.deleteItems(at: [indexPath])
-            } completion: { _ in
-                self.savedPaymentMethods.removeAll(where: {
-                    $0.stripeId == paymentMethod.stripeId
-                })
-
-                if let index = self.selectedViewModelIndex {
-                    if indexPath.row == index {
-                        self.selectedViewModelIndex = min(1, self.viewModels.count - 1)
-                    } else if indexPath.row < index {
-                        self.selectedViewModelIndex = index - 1
-                    }
-                }
+        guard let indexPath = collectionView.indexPath(for: paymentOptionCell),
+              case .saved(let paymentMethod) = viewModels[indexPath.row]
+        else {
+            assertionFailure()
+            return
+        }
+        let viewModel = viewModels[indexPath.row]
+        self.viewModels.remove(at: indexPath.row)
+        // the deletion needs to be in a performBatchUpdates so we make sure it is completed
+        // before potentially leaving edit mode (which triggers a reload that may collide with
+        // this deletion)
+        self.collectionView.performBatchUpdates {
```

Comment:
I'm seeing a problem where the pay button is disabled after deleting a payment method.
1. set `allowRemovalOfLastSavedPaymentMethod == false`
2.  add two payment methods (PM1, and PM2)
3. present payment sheet, tap edit
4. Delete PM2

Observed:
PM2 is deleted, edit mode is disabled, PM1 is selected, but the pay button is greyed out.

Reply:
Ah thanks for finding, looks like this is an extension of an existing bug. I have the fix but will put it in a separate follow-up PR.

Reply:
https://github.com/stripe/stripe-ios/pull/3258